include/denormals.h

```diff
@@ -4,27 +4,38 @@
 #ifndef DENORMALS_H
 #define DENORMALS_H
 
-
 #ifdef __SSE__
-#include <xmmintrin.h>
-#endif
-#ifdef __SSE3__
-#include <pmmintrin.h>
+#if defined(_MSC_VER)
+#include <immintrin.h>
+#elif defined(__GNUC__)
```

Comment:
How about other unusual compilers?

Reply:
OK. Let's assume any compiler missing from [Wikipedias compiler list](https://en.wikipedia.org/wiki/List_of_compilers#C++_compilers) is irrelevant enough to ignore. That is, until someone using it complains and gets to be the designated dev for that oddity.

Filtering the ones with a green C++17 column, it looks to me that all should be covered. (C++ Builder is LLVM, i.e. clang, EDG makes a front end from what I can tell, not a compiler, and then there's the GCC/Clang/Visual C++ trio). 

Out of the four partially supporting ones, two are for embedded systems (IAR and SEGGER) and Intel is LLVM. Which leaves us with the PGI compiler, now part of the NVidia HPC SDK, which I'm not sure of. Will have to return to that later.

Reply:
How about always including `immintrin.h` and additionally including `x86intrin.h` for GCC for extra safety?